### PR TITLE
feat: add migration for authorization list

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Closes:https://github.com/zeta-chain/node/issues/2311
Closes: #2376
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

## Summary by CodeRabbit

- **New Features**
  - Added a migration script for adding an authorizations table, enhancing the database schema functionality.
  - Implemented a new migration process to update the authority module from version 1 to version 2.

- **Bug Fixes**
  - Updated the `ConsensusVersion` to ensure proper versioning and avoid conflicts during migration.

- **Tests**
  - Added tests to validate the migration of the authorization list and default authorizations list.

- **Documentation**
  - Updated the changelog to reflect recent changes and improvements.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -236,7 +236,7 @@ start-stress-test: zetanode
 #TODO: replace OLD_VERSION with v16 tag once its available
 zetanode-upgrade: zetanode
 	@echo "Building zetanode-upgrade"
-	$(DOCKER) build -t zetanode:old -f Dockerfile-localnet --target old-runtime --build-arg OLD_VERSION='release/v16' .
+	$(DOCKER) build -t zetanode:old -f Dockerfile-localnet --target old-runtime --build-arg OLD_VERSION='release/v17' .
```
Comment (kingpinXD) [inline Makefile:239]: The upgrade test should trigger the migration now when we create a release. 

I'm unsure of the best way to test an upgrade when merging migration scripts other than unit tests. Any ideas are welcome

**Review by swift1337 — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (lumtis) [reply Makefile:239]: > other than unit tests

You mean E2E tests?

Comment (lumtis) [reply Makefile:239]: With E2E tests, I think we might need to consider a Make command similar to
```
start-upgrade-test-light
```
but running admin tests instead 
```
start-upgrade-test-admin
```

**Review by lumtis — APPROVED** [review]
Approving, my suggestion would be a futther PR

**Review by gartnera — APPROVED** [review]

**Pull request merged**